src/create_disk.sh

```diff
@@ -80,17 +80,46 @@ set -x
 # Partition and create fs's. The 0...4...a...1 uuid is a sentinal used by coreos-gpt-setup
 # in ignition-dracut. It signals that the disk needs to have it's uuid randomized and the
 # backup header moved to the end of the disk.
-sgdisk -Z $disk \
-	-U 00000000-0000-4000-a000-000000000001 \
-	-n 1:0:+384M -c 1:boot \
-	-n 2:0:+127M -c 2:EFI-SYSTEM -t 2:C12A7328-F81F-11D2-BA4B-00A0C93EC93B \
-	-n 3:0:+1M   -c 3:BIOS-BOOT  -t 3:21686148-6449-6E6F-744E-656564454649 \
-	-n 4:0:0     -c 4:root       -t 4:4F68BCE3-E8CD-4DB1-96E7-FBCAF984B709
-sgdisk -p "$disk"
+case "$arch" in
+    x86_64|aarch64)
+        sgdisk -Z $disk \
+        -U 00000000-0000-4000-a000-000000000001 \
+        -n 1:0:+384M -c 1:boot \
+        -n 2:0:+127M -c 2:EFI-SYSTEM -t 2:C12A7328-F81F-11D2-BA4B-00A0C93EC93B \
+        -n 3:0:+1M   -c 3:BIOS-BOOT  -t 3:21686148-6449-6E6F-744E-656564454649 \
+        -n 4:0:0     -c 4:root       -t 4:0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        sgdisk -p "$disk"
+        BOOTPN=1
+        EFIPN=2
+        BIOSPN=3
+        ROOTPN=4
+        ;;
+    s390x)
+        sgdisk -Z $disk \
+        -U 00000000-0000-4000-a000-000000000001 \
+        -n 1:0:+384M -c 1:boot \
+        -n 2:0:0     -c 2:root       -t 2:0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        sgdisk -p "$disk"
+        BOOTPN=1
+        ROOTPN=2
```

Comment:
I may have missed this previously, but lets keep root always on 4 and boot always on 1, unless there's technical reasons we cant.

Reply:
I have never used gaps in the partition numbering. Are there any implications to having gaps in partition numbers?

Reply:
Not that I know. Container Linux has gaps. They're just indices into the partition table. Empty entries are just zeroed out IIRC and the table size is fixed. Linux will create block devices correctly (e.g. have `/dev/vda1` and `/dev/vda4` even if 2 and 3 don't exist).

Reply:
Also worth noting the indices don't have to match the partition order physically on disk. You can have a disk laid out like:
```
[ 1 ][ 4 ][ 2 ][ 3 ]
```
And that's fine.

Reply:
Ok, thanks. I just wanted to ask about that as that is not something that I have seen in wild or used myself.
Done